docs/features/network-security-controls/admin-network-policy.md

```diff
@@ -1,303 +1,1118 @@
-    - namespaces:
-        namespaceSelector:
+    - action: Allow
+      to:
+      - namespaces:
+          matchLabels:
+            tenant: blue
+---
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: AdminNetworkPolicy
+metadata:
+  name: red-tenant
+spec:
+  priority: 40
+  subject:
+    namespaces: 
+      matchLabels:
+        tenant: red
+  ingress:
+    - action: Allow
+      from:
+      - namespaces:
+          matchLabels:
+            tenant: red
+  egress:
+    - action: Allow
+      to:
+      - namespaces:
+          matchLabels:
+            tenant: red
+---
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: AdminNetworkPolicy
+metadata:
+  name: yellow-tenant
+spec:
+  priority: 40
+  subject:
+    namespaces: 
+      matchLabels:
+        tenant: yellow
+  ingress:
+    - action: Allow
+      from:
+      - namespaces:
+          matchLabels:
+            tenant: yellow
+  egress:
+    - action: Allow
+      to:
+      - namespaces:
+          matchLabels:
+            tenant: yellow
+---
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: AdminNetworkPolicy
+metadata:
+  name: green-tenant
+spec:
+  priority: 40
+  subject:
+    namespaces: 
+      matchLabels:
+        tenant: green
+  ingress:
+    - action: Allow
+      from:
+      - namespaces:
+          matchLabels:
+            tenant: green
+  egress:
+    - action: Allow
+      to:
+      - namespaces:
           matchLabels:
-            conformance-house: slytherin
+            tenant: green
+---
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: AdminNetworkPolicy
+metadata:
+  name: multi-tenant-isolation
+spec:
+  priority: 43
+  subject:
+    namespaces: 
+      matchExpressions:
+        - {key: tenant, operator: In, values: [red, blue, green, yellow]}
+  ingress:
+    - action: Deny
+      from:
+      - namespaces: {}
+  egress:
+    - action: Deny
+      to:
+      - namespaces: {}
 ```
 
-* BANP doesn't have any priority field, since we can have only one in the cluster
-* We keep nbdb.ACL's priority range 1750 - 1649 range reserved for BANP rules in the cluster.
-* Rest of the implementation details for ANP is applicable to BANP as well.
+In order to be able to express "allow-to-yourself"
+and "deny-from-everyone-else" better, we have some ongoing
+upstream design work where we hope to be able to condense
+1 policy per tenant to be 1 policy for all tenants. Until
+then unfortunately you'd need to create 1 policy per tenant
+to achieve isolation.
+
+## Troubleshooting
 
-Corresponding ACLs:
+In order to ensure the ANP and BANP are correctly created,
+do `kubectl get anp` or `kubectl get banp` to see the status
+of the CRs. A sample good status will look like this:
 
+```shell
+  Conditions:
+    Last Transition Time:  2024-06-08T20:29:00Z
+    Message:               Setting up OVN DB plumbing was successful
+    Reason:                SetupSucceeded
+    Status:                True
+    Type:                  Ready-In-Zone-ovn-control-plane
```

Comment:
Oof... Having programmatically-generated Type names is not idiomatic kubernetes.

Note that `NetworkPolicy.Status` was abandoned because there didn't seem to be anything useful you could do with it

Reply:
yeah this one of those IC zone things ultimately; we felt there was value in knowing which zone went wrong if any in applying the policy